app/test/controllers/Users/store.js

```diff
@@ -0,0 +1,131 @@
+const { expect } = require('chai');
+
+const Register = require('../../helpers/register');
+const UserFactory = require('../../factories/user');
+const truncateDatabase = require('../../helpers/truncate');
+
+const app = require('../../../index');
+const request = require('supertest')(app);
+
+let users = [];
+let loggedUserId = null;
+let loggedUserToken = null;
+
+describe('Users', () => {
+    before(async () => {
+        await truncateDatabase();
+
+        const { user, token } = await Register(request);
+        loggedUserToken = token;
+        loggedUserId = user.id;
+
+        users.push(await UserFactory.create());
+        users.push(await UserFactory.create());
+        users.push(await UserFactory.create());
+    });
+
+    describe('POST /users', () => {
+        it('registers a new user when passing valid data', async () => {
+            const userData = UserFactory.generate();
+
+            let response = await request.post(`/users`).send(userData);
+
+            expect(response.body).to.have.property('auth').to.equal(true);
+        });
+
+        it('returns an error if firstName is blank', async () => {
```

Comment:
one test for all blank params - like our standard

in some futer we need hardly refactor our tests